app.py

```diff
@@ -44,4 +44,153 @@ def show_breakdown_dialog(self):
+        
+        # get list of breakdown items
+        items = breakdown_app.analyze_scene()
+        
+        # now loop over all items
+        for item in items:
+        
+            # get the latest version on disk
+            latest_version = breakdown_app.compute_highest_version(item["template"], item["fields"])
+            
+            # if our current version is out of date, update it!
+            current_version = item["fields"]["version"]
+            if latest_version > current_version:
+                
+                # make a fields dictionary representing the latest version
+                latest_fields = copy.copy(item["fields"])
+                latest_fields["version"] = latest_version
+                
+                # request that the breakdown updates to the latest version
+                breakdown_app.update_item(item["node_type"], item["node_name"], item["template"], latest_fields)
+
+        
+        
+        :returns: List of dictionaries, see above for example.
+        """
+        tk_multi_breakdown = self.import_module("tk_multi_breakdown")
         
+        # first, scan the scene and get a list of items
+        items = tk_multi_breakdown.get_breakdown_items()
+        
+        # if shotgun data is returned for an item, trim this down
+        # to return a more basic listing than the one returned
+        # from get_breakdown_items:
+        for item in items:
+            
+            if item["sg_data"]:
+                new_sg_data = {}
+                new_sg_data["id"] = item["sg_data"]["id"]
+                new_sg_data["type"] = item["sg_data"]["type"]
+                new_sg_data["code"] = item["sg_data"]["code"]
+                new_sg_data["task"] = item["sg_data"]["task"]
+                new_sg_data["name"] = item["sg_data"]["name"]
+                new_sg_data["entity"] = item["sg_data"]["entity"]
+                new_sg_data["project"] = item["sg_data"]["project"]
+                new_sg_data["version_number"] = item["sg_data"]["version_number"]
+                new_sg_data["published_file_type"] = item["sg_data"]["published_file_type"]
+                item["sg_data"] = new_sg_data
+                
+        return items
+
+
+    def compute_highest_version(self, template, fields):
+        """
+        Given a template and some fields, return the highest version number found on disk.
+        The template key containing the version number is assumed to be named {version}.
+        
+        This will perform a scan on disk to determine the highest version.
+        
+        For a usage example, see the analyze_scene() method.
+         
+        :param template: Template object to calculate for
+        :param fields: A complete set of fields for the template
+        :returns: The highest version number found
+        """
+        tk_multi_breakdown = self.import_module("tk_multi_breakdown")
+        return tk_multi_breakdown.compute_highest_version(template, fields)
```

Comment:
Wouldn't a failed publish (the file is created but not registered with Shotgun) be returned if it was the highest version, resulting in a false positive?